akka-stream/src/main/scala/akka/stream/impl/Sinks.scala

```diff
@@ -4,22 +4,21 @@
 package akka.stream.impl
 
 import java.util.concurrent.atomic.AtomicReference
-
 import akka.actor.{ ActorRef, Props }
 import akka.stream.impl.StreamLayout.Module
 import akka.stream.scaladsl.OperationAttributes
 import akka.stream.{ Inlet, Shape, SinkShape }
 import org.reactivestreams.{ Publisher, Subscriber, Subscription }
-
 import scala.annotation.unchecked.uncheckedVariance
 import scala.concurrent.{ Future, Promise }
+import akka.stream.impl.ActorFlowMaterializerImpl.CreateContext
 
 /**
  * INTERNAL API
  */
 private[akka] abstract class SinkModule[-In, Mat](val shape: SinkShape[In]) extends Module {
 
-  def create(materializer: ActorFlowMaterializerImpl, flowName: String): (Subscriber[In] @uncheckedVariance, Mat)
+  def create(context: CreateContext): (Subscriber[In] @uncheckedVariance, Mat)
```

Comment:
this is still tied to ActorFlowMatImpl, but at least it paves the way for generalization